### PR TITLE
Conversion to Python 3

### PR DESCRIPTION
**Pull request by frstp64**
I have seen your tweet about python 3 support. Everything worked fine with 2to3. I am not too sure if the two lines of code in flatpandoc.py I needed to comment are necessary since it seems to work fine without them.

I could make the code Python 2 and Python 3 compatible, but it would bit a bit cumbersome, mostly because of urllib.parse.

### CODE REVIEW

**Comment by apas** [comment]
Thanks. I can't test right now, but I think [this](https://github.com/apas/athena/pull/9/commits/5fd8b163bba9c8c0c60cb5a15f3b3ba072f0be87#diff-9de40ef13f0271a46b14b8c27bbf004dR47) line needs to be reviewed. Python 3 paths are different on Linux kernel-based systems (eg: Ubuntu ships with 3 by default,) whereas on macOS it's in `/usr/local/bin/python3[.x]` if you install with `brew` (which i) is the most common way to do so, and ii) unless the user has specified a custom Brew path outside of `/usr/local/bin`.) We might want to check with `env` somehow.

**Comment by apas** [comment]
Reading through the [`virtualenv` docs](https://virtualenv.pypa.io/en/stable/reference/#cmdoption-p) I think the solution is to replace the absolute path in `virtualenv`'s `--python=` parameter with the python version we want, i.e.:

```
proc.check_call(["virtualenv", "--python=python3", "env"])
```

`virtualenv`, then, handles the path lookup by itself. I tested this in an Ubuntu machine (with the default `python3` in `/usr/bin/python3` version 3.5.2) and macOS (with `python3` via `brew` in `/usr/local/bin` version 3.6.5.)

**Comment by apas** [comment]
>  I am not too sure if the two lines of code in flatpandoc.py I needed to comment are necessary

Yeah, same here. Original author committed it [as a Unicode fix](https://github.com/fhirschmann/Flask-FlatPages-Pandoc/commit/4138a452bfb097d88f5273c96f8f7f7600d576dc). However, I tested with a German umlaut and an emoji in terminal Vim 8.1 on macOS 10.13.5 and it worked fine. I guess we can leave it commented out.

emoji

OK, this looks good. I tested with Python 3 on macOS with both Pandoc 1.19.x and 2.0.1 running athena live as well as building a static instance. I didn't come across any issues. I think I can merge. However, couple minor issues before I do so:

- In line 25 of `readme.md` put `Python 3` (in ticks so it's printed as code like the rest) in the beginning of the list and append a period at the end;
- In line 220 of `readme.md` append a \` at the end of `push` (it's a typo I missed when I updated the file;)
- Append after line 15 in `flatpandoc.py`: `- Support of Python 3 by @frstp64`.

I think these additional changes cover it.

Finally, please let me know where and how you tested before I can merge.

**Comment by frstp64** [comment]
Testing environments:
* Ubuntu 16.04 LTS on an Amazon AWS instance. pandoc 1.16.0.2 and Python 3.5.2
* Arch Linux on an armv7l architecture (Raspberry Pi 3). pandoc 2.2.1 and Python 3.6.5
* Manjaro 17.1.10. pandoc 2.2.1 and Python 3.6.5

I disabled some modules (pandoc-sidenote and pandoc-crossref) and on the arm architecture machine because trying to install them using cabal did not work.

File: readme.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ tight integration of graphics with text, and carefully chosen typography.
 ## Install
 
 Dependencies: `pandoc`, `pandoc-citeproc`, `pandoc-crossref`,
-`pandoc-sidenote`, Python 3
+`pandoc-sidenote`, `Python 3`
```
Comment (apas) [inline readme.md]: `Python  3` should be moved in the beginning of the list.

File: readme.md [hunk 2]
```diff
@@ -217,7 +217,7 @@ If you're using your own hosting solution you know what to do now. Happy
 blogging!
 
 For GitHub pages you can `init` a new git repo from within the `/build`
-directory since athena's ignoring it and `push from there.
+directory since athena's ignoring it and `push from there.`
```
Comment (apas) [inline readme.md]: Should be "`push` from there." not "`push from there.`"

**Review by apas — CHANGES_REQUESTED** [review]
See comments inline. Thanks.

**Comment by apas** [comment]
> Testing environments:
> 
> Ubuntu 16.04 LTS on an Amazon AWS instance. pandoc 1.16.0.2 and Python 3.5.2 \
> Arch Linux on an armv7l architecture (Raspberry Pi 3). pandoc 2.2.1 and Python 3.6.5 \
> Manjaro 17.1.10. pandoc 2.2.1 and Python 3.6.5

This looks thorough. As soon as you fix the minor documentation issues I can merge. Thanks for your work! Appreciate this.

File: readme.md (hunk 1 above)
Comment (apas) [reply readme.md]: Also a period in the end of the list. I.e.,

> Dependencies: `Python 3`, `pandoc`, `pandoc-citeproc`, `pandoc-crossref`,
> `pandoc-sidenote`.

**Pull request merged**